worlds/metroidprime/data/RoomData.py

```diff
@@ -0,0 +1,463 @@
+from dataclasses import dataclass, field
+from typing import Any, Callable, Dict, List, Optional
+import typing
+
+from BaseClasses import (
+    CollectionState,
+    ItemClassification,
+    LocationProgressType,
+    Region,
+)
+from ..BlastShieldRando import BlastShieldType
+from ..DoorRando import DoorLockType
+from ..Items import ProgressiveUpgrade, SuitUpgrade
+from ..Logic import (
+    can_beam_combo,
+    can_bomb,
+    can_charge_beam,
+    can_ice_beam,
+    can_missile,
+    can_plasma_beam,
+    can_power_beam,
+    can_power_bomb,
+    can_super_missile,
+    can_wave_beam,
+)
+from ..data.AreaNames import MetroidPrimeArea
+from ..Locations import MetroidPrimeLocation, every_location
+from .RoomNames import RoomName
+from .Tricks import TrickInfo
+from .DoorData import DoorData
+
+if typing.TYPE_CHECKING:
+    from .. import MetroidPrimeWorld
+
+
+def get_config_item_text(world: "MetroidPrimeWorld", location: str) -> str:
+    loc = world.multiworld.get_location(location, world.player)
+    assert loc.item
+    if not loc.item.player:
+        return "Nothing"
+    player_name = (
+        f"{world.multiworld.player_name[loc.item.player]}'s "
+        if loc.item.player != world.player
+        else ""
+    )
+    player_name = player_name.replace("&", "[and]")
+    item_name = loc.item.name.replace("&", "[and]")

```

Comment:
From what I understand `&` gets handled differently since it is used in string formatting for prime but other ones appear to render just fine. I'll have to do some testing around lengthy names though 🤔

Reply:
Confirmed that this works as expected, it'll eventually clip the message after 100 or so characters. It just renders anything it doesn't understand as a square